src/Lecture1.hs

```diff
@@ -123,4 +137,13 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = show n ++ " is greater than " ++ show (head counts) ++ " elements and lower than " ++ show (last counts) ++ " elements"
+    where
+        go :: [Int] -> [Int] -> [Int]
+        go result l
+            | null l = result
+            | n > head l = go [succ (head result), last result] (tail l)
+            | n < head l = go [head result, succ (last result)] (tail l)
+            | otherwise = go result (tail l)
+        counts = go [0, 0] list
```

Comment:
You can pass these two numbers as two separate arguments. This way you won't need to call `head` and `last` on every recursive call

Reply:
I need to return them in a tuple then, correct?

Reply:
You have several options here:

* Return a two-element list as you do now but only in the end
* Return a tuple to guarantee that the function returns exactly 2 elements (however, I haven't explained tuples in lecture 1)
* Return `String`: when you reach the end of recursion, you can construct the string as the result. This way you don't need to return numbers.